core/src/test/scala/com.mediative.sparrow/DataFrameReaderTest.scala

```diff
@@ -173,51 +195,107 @@ class DataFrameReaderTest extends FreeSpec with BeforeAndAfterAll {
+        val rdd1 = rdd1Maybe.toOption.get
+        assertResult(0) { rdd0.subtract(rdd1).count }
+        assertResult(0) { rdd1.subtract(rdd0).count }
+      }
 
-      assert(df.toRDD[TestToRdd1].toOption.get.first == expected)
-    }
+      import TestCaseClasses._
 
-    "round-trip an object containing an Int and an optional Int from RDD to DataFrame back to RDD" - {
-      "when containing some value" in {
-        import TestCaseClasses.TestToRdd2
+      "Int, String" in {
+        testRDD_DataFrame_codec(TestToRdd1(1, "a"))
+      }
 
-        val expected = TestToRdd2(1, Option(1))
-        val df =
-          sqlContext.createDataFrame(sc.parallelize(List(expected)))
+      "Int, Option[Int]" - {
+        "when Some(Int)" in {
+          testRDD_DataFrame_codec(TestToRdd2(1, Option(1)))
+        }
 
-        assert(df.toRDD[TestToRdd2].toOption.get.first == expected)
+        "when None" in {
+          testRDD_DataFrame_codec(TestToRdd2(1, Option.empty))
+        }
       }
 
-      "when containing no value" in {
-        import TestCaseClasses.TestToRdd2
-
-        val expected = TestToRdd2(1, Option.empty)
-        val df =
-          sqlContext.createDataFrame(sc.parallelize(List(expected)))
+      "String, java.sql.Timestamp" in {
+        pendingUntilFixed {
+          // FIXME:
+          // rdd1Maybe.isSuccess was false Failure(NonEmptyList(The field 'timestampVal' isn't a LongType as expected, TimestampType received.)) (DataFrameReaderTest.scala:203)
+          testRDD_DataFrame_codec(
+            TestToRdd3("a", Timestamp.valueOf("2015-07-15 09:00:00"))
+          )
+        }
+      }
 
-        assert(df.toRDD[TestToRdd2].toOption.get.first == expected)
+      "Int, Double" in {
+        pendingUntilFixed {
+          // FIXME:
+          // "org.apache.spark.SparkException: Job aborted due to stage failure"
+          // Caused by: java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.Integer
+          // at scala.runtime.BoxesRunTime.unboxToInt(BoxesRunTime.java:106)
+          testRDD_DataFrame_codec(TestToRdd4(1, 2.0))
+        }
       }
-    }
 
-    "round-trip an object containing a String and a Timestamp from RDD to DataFrame back to RDD" in {
-      import TestCaseClasses.TestToRdd3
+      "Int, Option[Double]" - {
+        "when Some(Double)" in {
+          testRDD_DataFrame_codec(TestToRdd5(1, Some(2.0)))
+        }
+        "when None" in {
+          testRDD_DataFrame_codec(TestToRdd5(1, None))
+        }
+      }
 
-      val expected =
-        TestToRdd3("a", Timestamp.valueOf("2015-07-15 09:00:00"))
+      "Int, Wrapped[String]" in {
+        pendingUntilFixed {
+          // FIXME:
+          // rdd1Maybe.isSuccess was false Failure(NonEmptyList(The field 'wrappedStringVal' isn't a StringType as expected, StructType(StructField(unwrap,StringType,true)) received.)) (DataFrameReaderTest.scala:207)
+          testRDD_DataFrame_codec(TestToRdd8(1, Wrap.Wrapped("foo")))
```

Comment:
This isn't a bug of Sparrow. I've explain the issue to Luis before, the issue here is that the data is saved using a different schema than how it's read. When the data is saved using `createDataFrame`, there's no possibility to tweak the structure, so if there's a case class that's just a wrapper, it will be saved in the same structure. However when using Sparrow, if a FieldConverter is used that expect a flat structure, then we have a clash. The solution is to defined [Row|Field]Converter for the same structure as will be used by Parquet so that it doesn't clash (or have Sparrow support conversions in both directions).